python/create_sbovl_ifg.py

```diff
@@ -38,25 +38,57 @@
 soi_path = os.path.join(GEOC_folder, pair, pair + '_soi_adf_scaled.geo.tif')
 output_path = os.path.join(GEOC_folder, pair, pair + '.geo.sbovldiff.adf.mm.tif')
 
+# Coherence paths
+boi_coh_path = os.path.join(GEOC_folder, pair, pair + '.geo.bovldiff.adf.cc.tif')
+soi_coh_path = os.path.join(GEOC_folder, pair, pair + '_soi_adf_coh.geo.tif')
+output_coh_path = os.path.join(GEOC_folder, pair, pair + '.geo.sbovldiff.adf.cc.tif')
+
 # Open input GeoTIFF files
 if os.path.exists(boi_path):
     boi_mm = open_geotiff(boi_path, fill_value=np.nan)
 else:
     print(f"{boi_path} doesn't exist ,please run create_bovl_ifg.sh first!")
     sys.exit(1)
+    
 if os.path.exists(soi_path):
     soi = open_geotiff(soi_path, fill_value=np.nan)
 else:
     print(f"{soi_path} doesn't exist ,please run create_soi.py first!")
     sys.exit(1)
-    
+
+# Open coherences
+if os.path.exists(boi_coh_path):
+    boi_coh = open_geotiff(boi_coh_path, fill_value=np.nan)
+else:
+    print(f"{boi_coh_path} doesn't exist ,please run create_bovl_ifg.sh first!")
+    sys.exit(1)
+if os.path.exists(soi_coh_path):
+    soi_coh = open_geotiff(soi_coh_path, fill_value=np.nan)
+else:
+    print(f"{soi_coh_path} doesn't exist ,please run create_soi.py first!")
+    sys.exit(1)
+
+
 # Process the data
 soi_mm = soi * 1000
 super_sboi = boi_mm.copy()
+super_sboi[super_sboi==0]= np.nan
+soi_mm[soi_mm==0]= np.nan
 super_sboi[np.isnan(super_sboi)] = soi_mm[np.isnan(super_sboi)]
-super_sboi[np.isnan(super_sboi)] = np.nan
+
+# Process and scale coherence data
+super_sboi_coh = boi_coh.copy()
+super_sboi_coh[super_sboi_coh==0] = np.nan
+soi_coh[soi_coh==0]=np.nan
+super_sboi_coh[np.isnan(super_sboi_coh)] = soi_coh[np.isnan(super_sboi_coh)]
+
+# Scale coherence data to 0-255
+super_sboi_coh[super_sboi_coh==0] = np.nan 
```

Comment:
better to do opposite:
super_sboi_coh[np.isnan(super_sboi_coh)] = 0
and then (removing line 87) export for example to float tif, and then use os.system to just convert through gdal:
cmd = 'gdal_translate -ot Byte -scale 0 1 0 255 -co COMPRESS=DEFLATE -co PREDICTOR=2 -of GTiff -a_srs EPSG:4326 .................'
os.system(cmd)
(after line 92)